example/eblob_backend.c

```diff
@@ -127,7 +127,18 @@ static int blob_iterate_callback_common(struct eblob_disk_control *dc, int fd, u
 		}
 
 		data_offset += sizeof(struct dnet_ext_list_hdr);
-		size -= sizeof(struct dnet_ext_list_hdr);
+
+		/*
+		 * When record is not yet committed (it was allocated on disk via write_prepare and optional write_plain),
+		 * its @data_size is zero and removing this header size ends up with negative size converted back to
+		 * very large positive number (0xffffffffffffffd0)
+		 *
+		 * @data_offset above is safe, since actually eblob had allocated all needed space,
+		 * it just hasn't yet filled eblob_disk_control
+		 */
+		if (!(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+			size -= sizeof(struct dnet_ext_list_hdr);
```

Comment:
Apparently this test will (or has to) fail without the change too. If it doesn't, your test is wrong and has to be fixed.
This commit doesn't change logic of the key reading, it only changes the size reported by the iterator.

This change fixes only size of the entry to be reported during iteration time like this one:
node: address:port, key: 1ae593e6c54eeb97c0b9590e980afd500bc0f8352bf6aef60e671c093413c315c8950c3ba3a60e6b231ba7a792a13e3b2b22f052f3c8ef64352ceda29240479f, flags: 0, ts: 1437326697/750386000, keys: 70366/95987, status: 0, size: 18446744073709551568, data:

18446744073709551568 size is 0xffffffffffffffd0 - or minus sizeof(struct dnet_ext_list_hdr)